### PR TITLE
[docs] Config YML updates

### LINKED ISSUE #2120: Update and edit config YML file descriptions
Some things can be better explained, more irrelevant options for apm-server can be deemphasized, etc.

### PR DESCRIPTION
**Pull request by bmorelli25**
Closes #2120. 

* A lot of these changes are consistency related. Specifically, I made the headers the same size, ensured consistent spacing, and added periods where they were forgotten. I think they make the YML file look cleaner, but at the end of the day they aren't _super_ necessary, so I can undo them if you want to make the diff cleaner.
* Fixed some spelling/grammar issues
* Clarifications on what certain settings do.
* Fixed a few inconsistencies between the docs and the YML files. Most of these are also upstream in the Beats repo and will need to be persisted there.

@jalvz - perhaps we can chat about what you have in mind for irrelevant options to be deemphasized.

### CODE REVIEW

File: _meta/beat.yml [hunk 1]
```diff
@@ -108,21 +105,21 @@ apm-server:
     # The default pattern excludes stacktrace frames that have a filename starting with '/webpack'
     #exclude_from_grouping: "^/webpack"
 
-    # If a source map has previously been uploaded, source mapping is automatically applied
+    # If a source map has previously been uploaded, source mapping is automatically applied.
     # to all error and transaction documents sent to the RUM endpoint.
     #source_mapping:
 
-      # Sourcemapping is enabled by default for RUM events
+      # Sourcemapping is enabled by default for RUM events.
```
Comment (jalvz) [inline _meta/beat.yml]:
```suggestion
      # Sourcemapping is enabled by default.
```

(sourcemapping is *only* for RUM events)

File: _meta/beat.yml [hunk 2]
```diff
@@ -141,28 +138,30 @@ apm-server:
...
 
 
-  # If set to true, APM Server augments data received by the agent with the original IP of the backend server,
-  # or the IP and User Agent of the real user (RUM requests). It defaults to true.
+  # If true (default), APM Server captures the IP of the instrumented service
+  # or the IP and User Agent of the real user (RUM requests).
   #capture_personal_data: true
 
-  # golang expvar support - https://golang.org/pkg/expvar/
+  # Enable APM Server golang expvar support (https://golang.org/pkg/expvar/).
```
Comment (jalvz) [inline _meta/beat.yml]:
```suggestion
  # Enable APM Server Golang expvar support (https://golang.org/pkg/expvar/).
```

File: _meta/beat.yml [hunk 3]
```diff
@@ -368,10 +363,10 @@ output.elasticsearch:
   # The indices are all prefixed with `apm-%{[observer.version]}`.
   # To allow managing indices based on their age, all indices (except for sourcemaps)
   # end with the information of the day they got indexed.
-  # e.g. "apm-6.3.0-transaction-2018.03.20"
+  # e.g. "apm-7.3.0-transaction-2019.07.20"
```
Comment (jalvz) [inline _meta/beat.yml]: :clap: :clap:

**Comment by jalvz** [comment]
Thanks @bmorelli25 !

> what you have in mind for irrelevant options to be deemphasized

I actually don't remember, probably nothing :neutral_face: 
I just feel it is a lot to go trough for new users, maybe different sections (in the docs, not in the yml) for basic/advanced settings...? if you have any idea and feel creative, go for it :)

**Comment by bmorelli25** [comment]
Beats repo PR: https://github.com/elastic/beats/pull/13205

**Pull request merged**